### PR TITLE
transaction delay

### PR DESCRIPTION
**Pull request by MSusik**
Signed-off-by: Mateusz Susik mateusz.susik@cern.ch

### CODE REVIEW

**Comment by MSusik** [comment]
This implementation requires some changes in `wrap` method. The new instance of `JSONBase` should be discovered also when there is no schema, but the value is a dictionary or a list.

Regarding the solution with `multidict`. `multidict` is not a representation of multiple dictionaries, but a representation of a single dictionary with multiple keys. It is a difference. Why can it be a problem? For example:

``` python
with data.transaction_delayed as d:
    d['author'][0].append({'family_name': 'Smith'})
```

If there is an append function implemented in the `JSONList` class, the multidict will reuse the list that already exists there, and it will perform validation.

To achieve the goal, I think we need a full copy of the object. Here I included a simple implementation with such a copy.

File: jsonalchemy/dictstyle.py [hunk 1]
```diff
@@ -28,23 +30,55 @@ def wrap(value, value_schema, callbacks=None):
     return value
 
 
-class JSONDict(collections.Mapping):
+class JSONBase(object):
 
-    def __init__(self, mapping=None, schema=None, callbacks=None):
+    def __init__(self, schema, callbacks):
```
Comment (jirikuncar) [inline jsonalchemy/dictstyle.py]: `schema=None, callbacks=None):`

File: jsonalchemy/dictstyle.py [hunk 2]
```diff
@@ -89,22 +123,25 @@ def __delitem__(self, name):
             self.dict[name] = value
             raise e
 
+    def copy(self):
+        return JSONDict({self._copy_element(k): self._copy_element(v) for k, v
+                         in self.dict.iteritems()})
```
Comment (MSusik) [inline jsonalchemy/dictstyle.py]: Note to self: change to using six's iteritems.

**Comment by kaplun** [comment]
@MSusik can you make the PR on top of the official repo?

**Comment by MSusik** [comment]
Superseeded by 

https://github.com/inveniosoftware/jsonalchemy/pull/30

**Pull request closed without merging**